pkg/kubelet/kubelet.go

```diff
@@ -1974,6 +1975,18 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 	return true
 }
 
+func syncPod(kl *Kubelet, update proberesults.Update, handler SyncHandler) {
+	// We should not use the pod from manager, because it is never updated after initialization.
+	pod, ok := kl.podManager.GetPodByUID(update.PodUID)
+	if !ok {
+		// If the pod no longer exists, ignore the update.
+		klog.V(4).Infof("SyncLoop: ignore irrelevant update: %#v", update)
```

Comment:
it used to say "`(container unhealthy)`". Is this piece of information worth preserving?

Reply:
initially this code was only used when livenessProbe failed, but now that we use it to handle success of readiness and startup too, it's incorrect to report `(container unhealthy)`

Reply:
I understand, my point was that this extra message about container being unhealthy may be useful for `livenessProbe`. Maybe this can be added somehow. Or maybe add a type of a probe and status to the message?